x-pack/platform/plugins/shared/fleet/server/routes/preconfiguration/index.ts

```diff
@@ -19,8 +20,14 @@ export const registerRoutes = (router: FleetAuthzRouter) => {
     .post({
       path: PRECONFIGURATION_API_ROUTES.RESET_PATTERN,
       access: 'public',
-      fleetAuthz: {
-        fleet: { all: true },
+      security: {
+        authz: {
+          requiredPrivileges: [
+            FLEET_API_PRIVILEGES.AGENTS.ALL,
+            FLEET_API_PRIVILEGES.AGENT_POLICIES.ALL,
+            FLEET_API_PRIVILEGES.SETTINGS.ALL,
```

Comment:
Should this include integrations privilege too? 
https://github.com/elastic/kibana/blob/b2344549fbe5a57506c9bb7284623749a124bc06/x-pack/platform/plugins/shared/fleet/common/authz.ts#L112

Reply:
yes probably need at least `integration:read` 👍

Reply:
will it work if a user has only `integration:all`?

Reply:
Yes if a user has integration all it will also have the read permissions https://github.com/nchaulet/kibana/blob/76835ffa89687ae18fced37ed43cb7a6e4e3265b/x-pack/platform/plugins/shared/fleet/server/plugin.ts#L512